Update hosts.php

LiFiCode opened the pull request:
Add option to show only active host devices

LiFiCode committed:
Update hosts.php

Add option to show only active host devices

blacksenator commented:
Hello @LiFiCode, I appreciate your input. Giving out only the active devices is a good idea. Unfortunately, I don't really like the solution, because output is either a SimpleXMLElement or an array (of SimpleXMLElement). Please find a solution so that the same type is output in both cases.

Hint: https://stackoverflow.com/questions/9643116/deleting-simplexmlelement-node

Regards

Black Senator

LiFiCode commented:
You are totally right. 
Now outputs SimpleXMLElement instead of array.

I opted for creating a new SimpleXMLElement instead of removing  nodes from the original. The comment Node gets lost on the way - but it is empty anyway.

LiFiCode committed:
Update hosts.php

Now returns SimpleXMLElement instead of array.

LiFiCode committed:
Update hosts.php

Added filter by interface type

LiFiCode committed:
Update hosts.php

blacksenator commented on src/hosts.php:
```diff
@@ -356,14 +356,25 @@ public function getMeshList(bool $asJSON = false)
-        return simplexml_load_file($url);
+        $hostList = simplexml_load_file($url);
+        if ($onlyActive) {
+            $hostListxml = '<?xml version="1.0"?><Item>';
```
```
        $hostListXML = '<?xml version="1.0" encoding="UTF-8"?><List>';
            foreach ($hostList->xpath('Item[Active="1"]') as $host) {
                if (empty($interfaceType)
                    || (string)$host->InterfaceType == $interfaceType) {
                    $hostListXML .= $host->asXML();
                }
                return new \SimpleXMLElement($hostListXML . '</List>');
            }

        return $hostList;
```

blacksenator commented on src/hosts.php:
```diff
@@ -356,14 +356,25 @@ public function getMeshList(bool $asJSON = false)
+     * all known devices (default) or only currently active host devices optionally
+     * filtered by interface type.
      *
+     * @param bool $onlyActive
```
` * @param bool $active`

blacksenator commented on src/hosts.php:
```diff
@@ -356,14 +356,25 @@ public function getMeshList(bool $asJSON = false)
      * @return simpleXMLElement
      */
-    public function getHostList()
+    public function getHostList(bool $onlyActive = false, string $interfaceType = '')
```
`bool $active = false,`

blacksenator commented on src/hosts.php:
```diff
@@ -356,14 +356,25 @@ public function getMeshList(bool $asJSON = false)
-
-        return simplexml_load_file($url);
+        $hostList = simplexml_load_file($url);
+        if ($onlyActive) {
```
ìf (acticve) {`

LiFiCode committed:
Update hosts.php

LiFiCode commented:
Made the changes, thanks for your feedback and of course the project itself...

blacksenator commented on src/hosts.php:
```diff
@@ -360,20 +360,20 @@ public function getMeshList(bool $asJSON = false)
         $hostList = simplexml_load_file($url);
-        if ($onlyActive) {
+        if ($active) {
             $hostListxml = '<?xml version="1.0"?><Item>';
```
`$hostListxml = <?xml version="1.0" encoding="UTF-8"?><List>`

blacksenator commented on src/hosts.php:
```diff
@@ -360,20 +360,20 @@ public function getMeshList(bool $asJSON = false)
                 if(empty($interfaceType) || (string) $host->InterfaceType == $interfaceType)
                     $hostListxml .= $host->asXML();
-            return new \SimpleXMLElement($hostListxml.'</Item>');
+            return new \SimpleXMLElement($hostListxml . '</Item>');
```
`return new \SimpleXMLElement($hostListxml . '</List>');`

LiFiCode committed:
Update hosts.php

Looks very much cleaner, of course. But I didn't want to go over the line length limit...

> Overview. Code MUST follow a "coding style guide" PSR [PSR-1]. Code MUST use 4 spaces for indenting, not tabs. There MUST NOT be a hard limit on line length; the soft limit MUST be 120 characters; lines SHOULD be 80 characters or less.

LiFiCode commented:
Looks very much cleaner, of course. But I didn't want to go over the line length limit...

> ... Code MUST follow a "coding style guide" PSR [PSR-1]. Code MUST use 4 spaces for indenting, not tabs. There MUST NOT be a hard limit on line length; the soft limit MUST be 120 characters; lines SHOULD be 80 characters or less.

blacksenator commented:
Sorry, I don´t know what you try to tell me. I appreciate coding style PSR. The last thing I mentioned is that you used a wrong node. It has to be: 
```PHP
<?xml version="1.0" encoding="UTF-8"?>
<List>
    <Item>
    ...
    </Item>
</List>
```
and not
```PHP
<?xml version="1.0"?>
<Item>
    <Item>
    ...
    </Item>
</Item>

LiFiCode commented:
i really missed that, i thought you were talking about the split up xml...
gonna fix that now.

LiFiCode committed:
Update hosts.php

<List> instead of <Item> in output xml
